dev/source/docs/building-setup-windows10.rst

```diff
@@ -16,31 +16,105 @@ Upon completion, you will be able to build ArduPilot binaries and run the native
+--------------------------------
+When using SITL with default options or ``--console --map`` the installed XWindows application must be started first. If not, the following error message will appear when running :
 
-  You can run XWindows applications (graphical programs, including SITL) by installing `VcXsrv <https://sourceforge.net/projects/vcxsrv/>`_, `Cygwin X <https://x.cygwin.com/>`_ or `Xming <https://sourceforge.net/projects/xming/>`_ on Windows. You will need to set the DISPLAY variable to point to your display: in WSL it is 127.0.0.1:0, in WSL2 it will be the ip address of the PC's network port and can be set by using the code below. For more information regarding configuring your XLaunch Software , you can refer to `this resource <https://itservices.cas.unt.edu/software/windows-x-server-vcxsrv>`_. Also in WSL2 you may have to disable the firewall for public networks, or create an exception in order for VcXsrv to communicate with WSL2:
-                                                                          
 ::
 
-  export DISPLAY=$(cat /etc/resolv.conf | grep nameserver | awk '{print $2}'):0
+    [Errno 111] Connection refused sleeping
+
+Using VcXsrv
+++++++++++++
+#. Select display settings: "Multiple Windows" but it is your preference
+#. Select how to start clients: **Must** select **"Start no Client"**
+#. Extra settings: **Must** check **"Disable access control"**
+
+Code Editing in WSL Using VSCode
+---------------------------------------------
+VSCode can be installed on the Windows 10 machine to edit and compile files inside of WSL. VSCode automatically integrates upon installion with WSL.
+
+    * Do not install VSCode inside of WSL using for example apt-get.
+    * Be sure that the folder you are working on is opened within the WSL remote. This can be achieved by clicking the green button in the bottom left corner.
 
+Windows Terminal
+----------------
+Windows Terminal enables multiple tabs that can switch between many Linux terminals, PowerShell, and more. Directions to install Windows Terminal `are here <https://docs.microsoft.com/en-us/windows/terminal/get-started>`_.
 
-.. tip:: For code editing you can install VSCode inside WSL.
-   
-.. tip::
+    * You can change the default terminal when opening Windows Terminal by going to "Settings" and moving the Ubuntu section to be first in the ``list`` section of the ``settings.json`` file
+    * The default starting directory can be changed by following the example below
 
-    Windows 10 includes "Windows Defender" virus scanner. It will slow down WSL quite a bit. Disabling it greatly improves disk performance but increases your risk to viruses so disable at your own risk. Here is one of many resources/videos that show you how to disable it: https://www.youtube.com/watch?v=FmjblGay3AM
+    .. code-block:: json
+
+        {
+        "guid": "{07b52e3e-de2c-5db4-bd2d-ba144ed6c273}",
+        "hidden": false,
+        "name": "Ubuntu-20.04",
```

Comment:
should also give instructions for Ubunut18.04LTS since its supported for two more years and people like myself have peripherals that have no drivers in 20.04 and cant switch

Reply:
I reworded this since this for the most part it is autogenerated besides the starting directory bit.